src/cocotb/triggers.py

```diff
@@ -294,13 +283,10 @@ def __repr__(self) -> str:
         return f"{type(self).__qualname__}()"
 
 
-class ReadWrite(GPITrigger, metaclass=_ParameterizedSingletonGPITriggerMetaclass):
+@singleton
+class ReadWrite(GPITrigger):
     """Fires when the read-write portion of the simulation cycles is reached."""
```

Comment:
```suggestion
    """Fires when the read-write simulation phase is reached."""
```
(while we are here...)